src/etc/inc/services.inc

```diff
@@ -1012,7 +1013,7 @@ function services_dhcpd_zones($ddns_zones)
                     if (!empty($zone['ddnsdomainkeyname']) && !empty($zone['ddnsdomainkey'])) {
                         if (!in_array($zone['ddnsdomainkeyname'], $added_keys)) {
                             $dhcpdconf .= "key {$zone['ddnsdomainkeyname']} {\n";
-                            $dhcpdconf .= "  algorithm hmac-md5;\n";
+                            $dhcpdconf .= "  algorithm {$zone['ddnsdomainalgorithm']};\n";
```

Comment:
Could it be there is another stray hmac-md5 in this file that needs replacement?

Reply:
mmmh didn't find any more occurrences of ```hmac-md5```. have you any concrete line for a blind man?

Reply:
Can’t check without a laptop. It just seems odd there are two select boxes added but only one authorative line changed. I think Ad mentioned the same thing. Maybe it’s ok as is and one of the selects doesn’t do anything?